src/test/java/de/dhbw/karlsruhe/students/mailflow/integration/MailboxParsingServiceTest.java

```diff
@@ -0,0 +1,57 @@
+package de.dhbw.karlsruhe.students.mailflow.integration;
+
+import de.dhbw.karlsruhe.students.mailflow.core.application.email.MailboxRepository;
+import de.dhbw.karlsruhe.students.mailflow.core.application.email.parsing.MailboxParser;
+import de.dhbw.karlsruhe.students.mailflow.core.application.email.parsing.MailboxParsingService;
+import de.dhbw.karlsruhe.students.mailflow.core.application.email.parsing.MailboxParsingServiceException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.Mailbox;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.enums.MailboxType;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Address;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class MailboxParsingServiceTest {
+
+  @Test
+  void testNonExistentMailboxThrows() {
+    // Arrange
+    Address mailboxOwner = new Address("someUser", "someDomain.de");
+    MailboxRepository mockedRepository = (userAddress, type) -> Optional.empty();
+    MailboxParser mockedParser = content -> null;
+    MailboxParsingService service = new MailboxParsingService(mockedRepository, mockedParser);
+
+    // Assert
+    Assertions.assertThrows(
+
+        // Act
+        MailboxParsingServiceException.class, () -> service.getMailboxOfAddress(mailboxOwner));
+  }
+
+  @Test
+  void existentMailboxCorrectlyProvided(@TempDir File tempDir)
```

Comment:
nice usage! I didn't knew of this depenendency injection example and possibility to get a temporary directory on the fly using junit. :)